dehydrated-bigip-common

```diff
@@ -243,6 +263,50 @@ createClientSSLProfile() {
   return 0
 }
 
+#-------------------------------------------------------
+# Function: getCurrentCertSerialNumber()
+# gets the current serialNumber from a certificate assigned to a ClientSSLProfile on a BIGIP
+# $1 => name of client SSL profile
+# $2 => partition
+#-------------------------------------------------------
+getCurrentCertSerialNumber() {
+  SSLPROFILE=$(restCall "GET" "/tm/ltm/profile/client-ssl/~${2}~${1}")
+  log "getClientSSLProfile() SSLPROFILE: $SSLPROFILE"
+
+
+  if [ "`echo ${SSLPROFILE} | grep '"kind":"tm:ltm:profile:client-ssl:client-sslstate"'`x" != "x" ] ; then
+    # Profile already exists
+    # Get the certificate from the SSL profile
+	#Get the certReference -> link property from the SSLClientProfile, and strip the hostname and /mgmt part of the link
+    CERTLINK=$(echo $SSLPROFILE | jsonq '["certReference"]["link"]' | sed -E -s "s|https://[a-zA-Z0-9\-\.]*/mgmt/|/|" )
+	log "getClientSSLProfile() CERTLINK: $CERTLINK"
+    
+    if [ "`echo ${CERTLINK}x`" != "x" ] ; then
+	  #Certlink exits and is not empty
+      CERT=$(restCall "GET" "$CERTLINK")
+      log "getClientSSLProfile() CERT: $CERT"
+	  
+	  if [ "`echo ${CERT} | grep '"kind":"tm:sys:file:ssl-cert:ssl-certstate"'`x" != "x" ] ; then
+	    SERIAL=$(echo $CERT | jsonq '["serialNumber"]' | sed -E -s "s|:||g" )
+		log "getClientSSLProfile() SERIAL: $SERIAL"
+		echo $SERIAL
+	  else
+	    log "Cert not found or valid"
+		return 3
+	  fi
+    else
+      log "SSL cert link not found, return empty"
```

Comment:
@Sander-agfa , what is supposed to happen if the certReference doesn't exist in the current cert?  It just seems to exit with return code 2 here and I'm not sure how to remedy the situation.

Reply:
@hildebrau , I should expect the function to return code 2 to the caller, and not let the script exit entirely.
I'll test this further on my dev setup, though I had this tested already.